### PR TITLE
chore: Add sync-repo-settings.yaml config

### PR DESCRIPTION
**Pull request by lqiu96**
Fixes: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/issues/2801

Modeled the configurations from java-cloud-bom and google-cloud-java. Required checks pulled from the existing configurations set.

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - 'cla/google'
+      - 'unitTests (17)'
+      - 'unitTests (19)'
+      - 'unitTests (21)'
+      - 'releaseCheck'
+      - 'conventionalcommits.org'
+      - 'SonarCloud Code Analysis'
+    requiredApprovingReviewCount: 1
+    requiresCodeOwnerReviews: true
+    requiresStrictStatusChecks: false
```
Comment (meltsufin) [inline .github/sync-repo-settings.yaml:17]: What does that mean?

**Review by meltsufin — CHANGES_REQUESTED** [review]

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (lqiu96) [reply .github/sync-repo-settings.yaml:17]: I believe this requires that the PR branch be up to date and that we will have to rebase when we merge in multiple PRs together.

image

Comment (meltsufin) [reply .github/sync-repo-settings.yaml:17]: Got it. We should keep it as false.

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      - 'cla/google'
+      - 'unitTests (8)'
+      - 'unitTests (11)'
+      - 'unitTests (17)'
+      - 'releaseCheck'
+      - 'conventionalcommits.org'
+    requiredApprovingReviewCount: 1
+    requiresCodeOwnerReviews: true
+    requiresStrictStatusChecks: false
+permissionRules:
```
[the remark this answers is not in the document]
Comment (lqiu96) [reply .github/sync-repo-settings.yaml:56]: Updated this to cloud java team only for now.

**Review by meltsufin — APPROVED** [review]

File: .github/sync-repo-settings.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+rebaseMergeAllowed: true
+squashMergeAllowed: true
+mergeCommitAllowed: false
+branchProtectionRules:
+  - pattern: main
+    isAdminEnforced: true
+    requiredStatusCheckContexts:
```
[the remark this answers is not in the document]
Comment (burkedavison) [reply .github/sync-repo-settings.yaml:7]: Yes -we want all the native tests as required.

[the remark this answers is not in the document]
Comment (lqiu96) [reply .github/sync-repo-settings.yaml:7]: I can't seem to find an easy way to enable this for all the values in the matrix. Enabling this for `parallel-nativeTests` marks the parent (?) job as required which never runs:
image

For now, I'll just manually add in all the native tests and we can update this if we find a better way.

**Comment by meltsufin** [comment]
@lqiu96 These settings don't seem to be working. I don't see the checks being marked as required in PRs.

**Comment by lqiu96** [comment]
Taking a look.

**Pull request merged**